### PR TITLE
add GPTNeoXForSequenceClassification

### PR DESCRIPTION
**Pull request by Asugawara**
# What does this PR do?

This PR adds the GPTNeoX Seq Classification.
Would you be able to check them?

In advance, thank you for this cool OSS!

ref: https://github.com/huggingface/transformers/pull/11906

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?
@sgugger
@ArthurZucker
@younesbelkada

### CODE REVIEW

**Review by younesbelkada — APPROVED** [review]
Thanks a lot for this contribution! This looks clean to me

File: src/transformers/models/gpt_neox/modeling_gpt_neox.py [hunk 1]
```diff
@@ -730,3 +730,130 @@ def _reorder_cache(self, past_key_values, beam_idx):
...
+                logger.warning(
+                    f"{self.__class__.__name__} will not detect padding tokens in `inputs_embeds`. Results may be "
+                    "unexpected if using padding tokens in conjunction with `inputs_embeds.`"
+                )
+
+        pooled_logits = logits[torch.arange(batch_size, device=logits.device), sequence_lengths]
+
+        loss = None
+        if labels is not None:
+            if self.config.problem_type is None:
```
Comment (sgugger) [inline src/transformers/models/gpt_neox/modeling_gpt_neox.py:828]: Let's move the labels to the logits device here, so that this supports model parallelism.

**Review by sgugger — APPROVED** [review]
Thanks for adding this new model!

File: src/transformers/models/gpt_neox/modeling_gpt_neox.py (hunk 1 above)
Comment (Asugawara) [reply src/transformers/models/gpt_neox/modeling_gpt_neox.py:828]: Thank you for your review!
I'll move it!

I guess related to #22561 !

Comment (sgugger) [reply src/transformers/models/gpt_neox/modeling_gpt_neox.py:829]: Indeed!
